report/include/scp/report.h

```diff
@@ -352,10 +352,11 @@ struct ScLogger {
     if (::scp::get_log_verbosity(__VA_ARGS__) >= sc_core::SC_MEDIUM) \
     SCP_LOG(sc_core::SC_MEDIUM, __VA_ARGS__)
 //! macro for warning level output
-#define SCP_WARN(...)                                          \
-    ::scp::ScLogger<::sc_core::SC_WARNING>(__FILE__, __LINE__, \
-                                           sc_core::SC_MEDIUM) \
-        .type(__VA_ARGS__)                                     \
+#define SCP_WARN(...)                                             \
+    if (::scp::get_log_verbosity(__VA_ARGS__) >= sc_core::SC_LOW) \
```

Comment:
Why should this check for SC_LOW be added? If you want to switch off Warnings just set the log level to ERROR and you are done. Aside of this warnings will not be counted if verbosity is set to SC_LOW although even in the SystemC implementation of reporting verbosity will only taken into account if severity is SC_INFO

Reply:
I think the issue is that there was no way to distinguish between warnings and errors. (In the case somebody wants errors, but not warnings).

Reply:
As earlier mentioned: you set the reporting level to log::ERROR and you are done. Yes, the filtering is done in the report handler itself so it goes thru the entire stack. But if you have that many warnings that this makes a difference you are doomed anyways ;-)

Reply:
Thats not what we found.
The use case is that we would like to be able to choose (freely) the level of reporting for each 'unit' (sc_module, or named thing).
Hence, to be able to control this with CCI, the 'global' level has to be set to print _all_ messages. The back-end has to print everything.
Then the CCI value will be used to set the actual level for each unit. This patch adds a hierarchy to this, which makes this somewhat more convenient to set - for instance by setting a 'global' CCI value.
But then the CCI value must be able to select which level you want, which includes e.g. printing only errors. That didn't seem possible before.

Reply:
Good point. I could not imagine that you want to switch of warnings on a component basis since in our projects we treat warnings as errors.

Reply:
(I'm not saying treating warnings as errors is a bad idea - but :-) )